pkg/terraform/exec/plugins/vsphereprivate/resource_vsphereprivate_import_ova.go

```diff
@@ -365,22 +365,23 @@ func resourceVSpherePrivateImportOvaCreate(d *schema.ResourceData, meta interfac
 		Network: importOvaParams.Network.Reference(),
 	}}
 
-	var diskType types.OvfCreateImportSpecParamsDiskProvisioningType
-
-	if d.Get("disk_type") == "thin" {
-		diskType = types.OvfCreateImportSpecParamsDiskProvisioningTypeThin
-	} else if d.Get("disk_type") == "eagerZeroedThick" {
-		diskType = types.OvfCreateImportSpecParamsDiskProvisioningTypeEagerZeroedThick
-	} else {
-		diskType = types.OvfCreateImportSpecParamsDiskProvisioningTypeThick
+	cisp := types.OvfCreateImportSpecParams{
+		EntityName:     d.Get("name").(string),
+		NetworkMapping: networkMappings,
+	}
+	if d.Get("disk_type") != "" {
+		if d.Get("disk_type") == "thin" {
+			cisp.DiskProvisioning = string(types.OvfCreateImportSpecParamsDiskProvisioningTypeThin)
+		} else if d.Get("disk_type") == "thick" {
+			cisp.DiskProvisioning = string(types.OvfCreateImportSpecParamsDiskProvisioningTypeThick)
+		} else if d.Get("disk_type") == "eagerZeroedThick" {
+			cisp.DiskProvisioning = string(types.OvfCreateImportSpecParamsDiskProvisioningTypeEagerZeroedThick)
+		} else {
+			errors.Errorf("Disk provisioning type is not supported: %s.", d.Get("disk_type").(string))
+		}
 	}
```

Comment:
Building on the comment from @jinyunma about missing the return when the disk type is invalid, it would be easier to read if this used a switch.
```suggestion
	switch diskType := d.Get("disk_type"); diskType {
	case "":
		// omit DiskProvisioning to use the default
	case "thin":
		cisp.DiskProvisioning = string(types.OvfCreateImportSpecParamsDiskProvisioningTypeThin)
	case "thick":
		cisp.DiskProvisioning = string(types.OvfCreateImportSpecParamsDiskProvisioningTypeThick) 
	case "eagerZeroedThick":
		cisp.DiskProvisioning = string(types.OvfCreateImportSpecParamsDiskProvisioningTypeEagerZeroedThick)
	default:
		return errors.Errorf("Disk provisioning type %q is not supported.", diskType)
	}
```